src/npcmove.cpp

```diff
@@ -3890,14 +3916,16 @@ bool npc::consume_food()
         // consume doesn't return a meaningful answer, we need to compare moves
         if( best_food != nullptr ) {
             const time_duration &consume_time = get_consume_time( *best_food );
-            consumed = consume( item_location( *this, best_food ) ) != trinary::NONE;
+            consumed = consume( *best_food ) != trinary::NONE;
```

Comment:
Ooopsies, I thought that `player::consume(item &)` was an easier form of `player::consume(item_location)`, but I've now looked at the code after having my companion try to eat a maple pie with -1 portions remaining, and have discovered they're not the same after all. :)